src/client/Client.cc

```diff
@@ -4056,8 +4060,10 @@ int Client::_do_remount(void)
           "failed to remount (to trim kernel dentries): "
           "return code = " << r << dendl;
     }
-    bool should_abort = cct->_conf->get_val<bool>("client_die_on_failed_remount") ||
-        cct->_conf->get_val<bool>("client_die_on_failed_dentry_invalidate");
+    bool should_abort =
+      (cct->_conf->get_val<bool>("client_die_on_failed_remount") ||
+       cct->_conf->get_val<bool>("client_die_on_failed_dentry_invalidate")) &&
+      !(retry_on_error && (++retries_on_invalidate < max_retries));
```

Comment:
for the case retry_on_error == false, we should let test_dentry_handling() call ceph_assert()

Reply:
@ukernel Does your comment apply to master as well?

Reply:
The comment applies to master as well. If there's a fix/enhancement in this part, we should fix it in master as a separate PR.

Reply:
@ukernel @vshankar If we can get the master PR opened and merged within a reasonable time frame, then it would be simple to cherry-pick it into this PR?

Reply:
@ukernel -- any particular reason as of why `test_dentry_handling()` should call `ceph_assert()`?

`_do_remount()` would call `ceph_abort()` on remount failure when testing dentry invalidation.

Reply:
ok, please remove the ceph_assert() in test_dentry_handling

Reply:
sure -- I'll have that up for master and its backport.